packages/frontend/src/state/transactions/hooks.ts

```diff
@@ -0,0 +1,112 @@
+import BigNumber from 'bignumber.js'
+import { useQuery } from '@apollo/client'
+import { useEffect } from 'react'
+
+import { networkIdAtom, addressAtom } from '../wallet/atoms'
+import { addressesAtom, isWethToken0Atom } from '../positions/atoms'
+import { useEthPriceMap } from '../ethPriceCharts/atoms'
+
+import { TRANSACTIONS_QUERY, TRANSACTIONS_SUBSCRIPTION } from '@queries/uniswap/transactionsQuery'
+import { transactions_positionSnapshots } from '@queries/uniswap/__generated__/transactions'
+import { BIG_ZERO, TransactionType } from '@constants/index'
+
+export const useLiquidityTxHistory = () => {
+  const [networkId] = useAtom(networkIdAtom)
+  const [address] = useAtom(addressAtom)
+  const isWethToken0 = useAtomValue(isWethToken0Atom)
+  const ethPriceMap = useEthPriceMap()
+
+  const { squeethPool, oSqueeth, shortHelper, swapRouter, crabStrategy } = useAtomValue(addressesAtom)
+
+  const { subscribeToMore, data, refetch, loading, error, startPolling, stopPolling } = useQuery(TRANSACTIONS_QUERY, {
+    variables: {
+      poolAddress: squeethPool,
+      owner: address,
+      origin: address || '',
+      recipients: [shortHelper, address || '', swapRouter],
+      orderDirection: 'desc',
+    },
+    fetchPolicy: 'cache-and-network',
+  })
+
+  useEffect(() => {
+    subscribeToMore({
+      document: TRANSACTIONS_SUBSCRIPTION,
+      variables: {
+        poolAddress: squeethPool,
+        owner: address,
+        origin: address || '',
+        recipients: [shortHelper, address || '', swapRouter],
+        orderDirection: 'desc',
+      },
+      updateQuery(prev, { subscriptionData }) {
+        if (!subscriptionData.data) return prev
+        const positionSnapshots = subscriptionData.data.positionSnapshots
+        return {
+          positionSnapshots: positionSnapshots,
+        }
+      },
+    })
+  }, [address, crabStrategy, networkId, oSqueeth, shortHelper, squeethPool, swapRouter, subscribeToMore])
+
+  const addRemoveLiquidityTrans =
+    ethPriceMap &&
+    (data?.positionSnapshots || []).map(
+      (transaction: transactions_positionSnapshots, index: number, array: transactions_positionSnapshots[]) => {
+        const transactionDetails = {
+          squeethAmount: new BigNumber(isWethToken0 ? transaction.depositedToken1 : transaction.depositedToken0),
+          ethAmount: new BigNumber(isWethToken0 ? transaction.depositedToken0 : transaction.depositedToken1),
+          usdValue: BIG_ZERO,
+          timestamp: transaction.transaction.timestamp,
+          transactionType: TransactionType.ADD_LIQUIDITY,
+          txId: transaction.transaction.id,
+          ethPriceAtDeposit: BIG_ZERO,
+        }
+
+        const squeethDepositedAmount = new BigNumber(
+          isWethToken0 ? transaction.depositedToken1 : transaction.depositedToken0,
+        )
+        const ethDepositedAmount = new BigNumber(
+          isWethToken0 ? transaction.depositedToken0 : transaction.depositedToken1,
+        )
```

Comment:
If you define squeethDepositedAmount and ethDepositedAmount in the above transaction details, you can reuse it when defining transactionDetails.

Reply:
cuz they r only used in liquidity txs, not sure if we put them into txs details